### PR TITLE
chore: add missing get-deps and clean targets for api-ts-sdk

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

- add missing get-deps and clean targets for api-ts-sdk
- remove a checked in dependency
- update the makefile format to use substitutes

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/api-ts-sdk/Makefile [hunk 1]
```diff
@@ -35,7 +35,4 @@ build: build-swagger
 
 .PHONY: clean
 clean:
-	rm -r deps/ || true
```
Comment (hkang1) [inline webui/api-ts-sdk/Makefile:38]: **question:** genuinely curious - if `deps` doesn't exist, then the `rm` fails but the `|| true` allows the next command to run without failure, but would the error get printed to the stdout? Why choose `rm -r X || true` over `rm -rf X` I guess is the real question?

File: webui/Makefile [hunk 2]
```diff
@@ -31,7 +34,7 @@ build-proto:
 
 .PHONY: build-api-ts-sdk
 build-api-ts-sdk: build-proto
-	$(MAKE) -C api-ts-sdk build
+	$(MAKE) -C api-ts-sdk get-deps build
```
Comment (hkang1) [inline webui/Makefile:37]: ah, I'm guessing is this why `tsc` was not available when building `api-ts-sdk`?

File: webui/Makefile [hunk 3]
```diff
@@ -2,20 +2,23 @@
...
+fmt: fmt-react fmt-elm fmt-tests
+.PHONY: fmt-%
+fmt-%:
+	$(MAKE) -C $(subst -,/,$*) clean
+
+.PHONY: check
+check: check-react check-elm check-tests
+.PHONY: check-%
+check-%:
+	$(MAKE) -C $(subst -,/,$*) clean
```
Comment (hkang1) [inline webui/Makefile:21]: **question:** what is this black magic?

File: webui/Makefile [hunk 4]
```diff
@@ -2,20 +2,23 @@
...
+	$(MAKE) -C api-ts-sdk $@
+.PHONY: clean-%
+clean-%:
+	$(MAKE) -C $(subst -,/,$*) clean
+
+.PHONY: fmt
+fmt: fmt-react fmt-elm fmt-tests
+.PHONY: fmt-%
+fmt-%:
+	$(MAKE) -C $(subst -,/,$*) clean
```
Comment (hkang1) [inline webui/Makefile:15]: **question:** should this line be:
```
$(MAKE) -C $(subst -,/,$*) fmt
```

File: webui/Makefile (hunk 3 above)
Comment (hkang1) [inline webui/Makefile:21]: same as the question above, should `clean` be `check`?

File: webui/api-ts-sdk/Makefile [hunk 5]
```diff
@@ -35,7 +35,4 @@ build: build-swagger
...
-	rm -r node_modules/ || true
-	rm -r src/ || true
-	rm -r lib/ || true
+	rm -rf deps/ node_modules/ src/ lib/
```
Comment (hkang1) [inline webui/api-ts-sdk/Makefile:38]: so much easier to read this

File: webui/api-ts-sdk/Makefile (hunk 1 above)
Comment (hamidzr) [reply webui/api-ts-sdk/Makefile:38]: I had to lookup -f again the man page shows:
```      
   -f, --force
                ignore nonexistent files and arguments, never prompt
```

I think the difference would be in the never prompt portion.. or less likely if you don't have `rm` or mistype something

Screen Capture_select-area_20200723121827

File: webui/Makefile (hunk 2 above)
Comment (hamidzr) [reply webui/Makefile:37]: exactly but also clean was missing so lack of get-deps wasn't showing

File: webui/Makefile (hunk 3 above)
Comment (hamidzr) [reply webui/Makefile:21]: haha it is the same excluding the added clean line for `api-ts-sdk` in the clean target. 
this is to make it look more like the project's main Makefile.

File: webui/Makefile (hunk 4 above)
Comment (hamidzr) [reply webui/Makefile:15]: yes it should! :facepalm:

**Comment by hamidzr** [comment]
> I was able to run `make all` after wiping the virtualenv and doing a `make clean` first.
> 
> I don't think it's blocking, but I had a couple of questions, not sure if there are typos or not. You understand the `Makefile` a lot better than I do.

thanks for the quick review and catching the copypasta error I made

File: webui/Makefile (hunk 2 above)
Comment (hamidzr) [inline webui/Makefile:37]: this part isn't parallelizable so probably we should have two lines instead of defining the dependency this way

**Comment by hamidzr** [comment]
@dzhu  any comments or improvements you see here?

File: webui/Makefile [hunk 6]
```diff
@@ -2,20 +2,23 @@
...
+fmt-%:
+	$(MAKE) -C $(subst -,/,$*) fmt
+
+.PHONY: check
+check: check-react check-elm check-tests
+.PHONY: check-%
+check-%:
+	$(MAKE) -C $(subst -,/,$*) check
 
 get-deps: build-api-ts-sdk
```
Comment (dzhu) [inline webui/Makefile:23]: Missing `.PHONY`.

File: webui/Makefile [hunk 7]
```diff
@@ -2,20 +2,23 @@
...
-	$(MAKE) -C elm $@
-	$(MAKE) -C tests $@
-
-check:
-	$(MAKE) -C react $@
-	$(MAKE) -C elm $@
-	$(MAKE) -C tests $@
+clean: clean-react clean-elm clean-tests
+	$(MAKE) -C api-ts-sdk $@
+.PHONY: clean-%
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply webui/Makefile:7]: I'll just leave it for now

File: webui/Makefile [hunk 8]
```diff
@@ -1,22 +1,27 @@
...
-	$(MAKE) -C tests $@
-
-check:
-	$(MAKE) -C react $@
-	$(MAKE) -C elm $@
-	$(MAKE) -C tests $@
-
+.PHONY: all
+all:
+	make get-deps
```
Comment (hamidzr) [inline webui/Makefile:3]: wat's the difference between make and $(MAKE)? it tells make that this is a submake? no upper makefile is calling this target thinking

Comment (dzhu) [reply webui/Makefile:3]: This should use `$(MAKE)`, which will allow this call to use the parallelism specified by the overall `make` invocation. So if you do `make -j4 all`, the `-j4` will be forwarded down into these.

File: webui/Makefile [hunk 9]
```diff
@@ -1,22 +1,27 @@
...
+clean: clean-react clean-elm clean-tests clean-api-ts-sdk
+.PHONY: clean-%
+clean-%:
+	$(MAKE) -C $* clean
+
+.PHONY: fmt
+fmt: fmt-react fmt-elm fmt-tests
+.PHONY: fmt-%
+fmt-%:
+	$(MAKE) -C $(subst -,/,$*) fmt
```
Comment (dzhu) [inline webui/Makefile:16]: Should probably get rid of the other uses of `subst` as well to keep it consistent.

File: webui/Makefile [hunk 10]
```diff
@@ -41,7 +47,8 @@ build-react:
 build-elm:
```
Comment (dzhu) [inline webui/Makefile:47]: At this point, we may as well go ahead and use the `build-%` pattern for `build-elm` and `build-react` too.

**Pull request merged**